Let users on email auth delete their mobile numbers

CrystalPea opened the pull request:
Sometimes users ask us to delete their mobile numbers for them.

If those users are on email auth, they should be able to delete their number themselves.

This will save them writing a support ticket and save us going into the database.

Here is the flow:

![Screenshot 2022-02-24 at 09 25 16](https://user-images.githubusercontent.com/20957548/155496913-c1063137-30dd-45c7-8fcd-cd8f923b7be5.png)

![Screenshot 2022-02-24 at 09 25 23](https://user-images.githubusercontent.com/20957548/155496917-9c3163ec-d3c1-4c5d-b413-1780255d3d7e.png)

![Screenshot 2022-02-24 at 09 28 04](https://user-images.githubusercontent.com/20957548/155497053-8c7dd6c6-bf0d-4d07-9091-01d471e73886.png)

benthorner commented on app/templates/views/user-profile/change.html:
```diff
@@ -20,9 +20,20 @@
       {% call form_wrapper() %}
         {{ form_field(error_message_with_html=True) }}
-        {{ page_footer('Save') }}
+        {% if current_user.auth_type == 'email_auth' %}
```
```suggestion
        {% if current_user.auth_type != 'sms_auth' %}
```

This is inconsistent with the conditional in the endpoint.

CrystalPea replied:
You are right it is inconsistent, I would change it the other way round - as people who use webauthn should still have their phone numbers as a backup for if they lose the key we downgrade them to sms auth if I remember the process around this correctly.

benthorner reviewed (commented):
Fair enough 👍.

I tried this out locally and noticed we still show "Delete your number":

- After I've deleted my number. 
- On the form to change my email.

Tweaking the conditional should fix the first one. We can also switch to using a dedicated template for the mobile number form so it won't leak into the other one.

CrystalPea commented:
@benthorner 

> I tried this out locally and noticed we still show "Delete your number":
>
> After I've deleted my number.
> On the form to change my email.
> Tweaking the conditional should fix the first one. We can also switch to using a dedicated template for the mobile number form so it won't leak into the other one.

Nice find 👏🏼 . I fixed it with a slightly clunky conditional, do you think it is ok?

benthorner commented on tests/app/main/views/test_user_profile.py:
```diff
@@ -183,6 +183,59 @@ def test_should_show_mobile_number_page(
+    assert 'Delete your number' not in page.text
+
+
+def test_change_your_mobile_number_page_shows_delete_link_if_user_on_email_auth(
```
Should we have a test to cover the other side of the conditional (not on email auth or doesn't have a mobile number setup)?

CrystalPea replied:
That's a good idea, I will do that

benthorner reviewed (approved):
Works fine locally for me now ✅.

I think the clunky conditional is OK - if anything else gets added in future then we should consider having a dedicated partial instead.
